### PR TITLE
refactor: Define EscrowFinish Transaction

### PR DESCRIPTION
**Pull request by natenichols**
## High Level Overview of Change
Defines `EscrowFinish` Transaction type

### Context of Change
Add `EscrowFinish` TypeScript definition, `verifyEscrowFinish` functionality,  and tests for `verifyEscrowFinish`

### Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### CODE REVIEW

**Review by mvadari — APPROVED** [review]

File: src/models/transactions/escrowFinish.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { ValidationError } from "../../common/errors";
+import { BaseTransaction, verifyBaseTransaction } from "./common";
+
+export interface EscrowFinish extends BaseTransaction {
+    TransactionType: "EscrowFinish";
+    Owner: string;
+    OfferSequence: number;
+    Condition?: string;
+    Fulfillment?: string;
```
Comment (mukulljangid) [inline src/models/transactions/escrowFinish.ts:9]: remove the semicolons and you're good to go!

File: src/models/transactions/transaction.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import Metadata from "../common/metadata";
 import { OfferCreate } from "./offerCreate";
 import { CheckCreate } from "./checkCreate";
-
+import { EscrowFinish } from "./escrowFinish";
```
Comment (mukulljangid) [inline src/models/transactions/transaction.ts:4]: ditto

**Review by mukulljangid — APPROVED** [review]
LGTM

**Review by khancode — CHANGES_REQUESTED** [review]
Suggesting the same changes to be made just like with [EscrowCancel PR](https://github.com/ripple/ripple-lib/pull/1511#pullrequestreview-733379253):
- make tests easier to write/read (using beforeEach())
- make error messages more concise
- nit: remove semicolons

**Review by khancode — APPROVED** [review]
LGTM!

**Pull request merged**